dev/io.openliberty.cdi.4.0.internal_fat/fat/src/io/openliberty/cdi40/internal/fat/bce/BuildCompatibleExtensionsTest.java

```diff
@@ -0,0 +1,167 @@
+    })
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        // basicwar.war
+        // ------------
+        Package warPackage = BasicBCETestServlet.class.getPackage();
+        WebArchive basicWar = ShrinkWrap.create(WebArchive.class, "basicWar.war")
+                                        .addPackage(warPackage)
+                                        .addAsServiceProvider(BuildCompatibleExtension.class, BasicBCEExtension.class)
+                                        .addAsWebInfResource(warPackage, "beans.xml", "beans.xml")
+                                        .addAsManifestResource(warPackage, "permissions.xml", "permissions.xml"); // Workaround WELD-2705
+
+        ShrinkHelper.exportDropinAppToServer(server, basicWar, DeployOptions.SERVER_ONLY);
+
+        // basicear.ear
+        // --------------
+        WebArchive war1 = ShrinkWrap.create(WebArchive.class, "war1.war")
+                                    .addPackage(EarBCETestServlet1.class.getPackage());
+
+        WebArchive war2 = ShrinkWrap.create(WebArchive.class, "war2.war")
+                                    .addPackage(EarBCETestServlet2.class.getPackage());
+
+        JavaArchive lib = ShrinkWrap.create(JavaArchive.class, "lib.jar")
+                                    .addPackage(LibTestBean.class.getPackage())
+                                    .addAsServiceProvider(BuildCompatibleExtension.class, EarBCEExtension.class)
+                                    .addAsManifestResource(LibTestBean.class.getPackage(), "beans.xml", "beans.xml");
+
+        EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, "basicear.ear")
+                                          .addAsModules(war1, war2)
+                                          .addAsLibraries(lib)
+                                          .addAsManifestResource(LibTestBean.class.getPackage(), "earPermissions.xml", "permissions.xml"); // Workaround WELD-2705
+
+        ShrinkHelper.exportDropinAppToServer(server, ear, DeployOptions.SERVER_ONLY);
+
+        // enhance.war
+        // -----------
+        Package enhancePackage = BceEnhanceTestServlet.class.getPackage();
+        WebArchive enhanceWar = ShrinkWrap.create(WebArchive.class, "enhance.war")
+                                          .addPackage(enhancePackage)
+                                          .addAsServiceProvider(BuildCompatibleExtension.class, EnhanceTestExtension.class)
+                                          .addAsWebInfResource(enhancePackage, "beans.xml", "beans.xml")
+                                          .addAsManifestResource(enhancePackage, "permissions.xml", "permissions.xml");
+
+        ShrinkHelper.exportDropinAppToServer(server, enhanceWar, DeployOptions.SERVER_ONLY);
+
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
```

Comment:
Would it be a good idea to separate the "good" tests from the "bad", put them on separate servers?